cranelift/isle/veri/veri_engine/src/solver/encoded_ops/cls.rs

```diff
@@ -0,0 +1,3500 @@
+use crate::solver::SolverCtx;
+use easy_smt::SExpr;
+
+// Adapted from https://stackoverflow.com/questions/23856596/how-to-count-leading-zeros-in-a-32-bit-unsigned-integer
+
+pub fn cls64(solver: &mut SolverCtx, x: SExpr, id: u32) -> SExpr {
+    // Generated code.
```

Comment:
perfectly fine to upstream this for now, but idle question: are there any thoughts or potential plans to have a "prelude" in native SMT-LIB that we feed to the solver, or some other way to avoid having to open-code these inlined algorithms?

Reply:
Added some comments to that effect. We would eventually like the express these in the annotation language, but it's not quite there yet (we also expect `subs` to not be needed once we have rule chaining).